scvi/inference/posterior.py

```diff
@@ -505,39 +506,58 @@ def imputation(self, n_samples=1):
         return imputed_list.squeeze()
 
     @torch.no_grad()
-    def generate(self, n_samples=100, genes=None):  # with n_samples>1 return original list/ otherwose sequential
+    def generate(
+        self,
+        n_samples: int = 100,
+        genes: Union[list, np.array] = None,
+        batch_size: int = 128
+    ):
         """
-        Return original_values as y and generated as x (for posterior density visualization)
-        :param n_samples:
-        :param genes:
-        :return:
+        Create observation samples from the Posterior Predictive distribution
+
+        :param n_samples: Number of required samples for each cell
+        :param genes: Indices of genes of interest
+        :param batch_size: Desired Batch size to generate data
+
+        :return: Tuple (x_new, x_old)
```

Comment:
python supports Typing for returned values, it is very useful when developing in smart IDEs like pycharm, please consider adding it here and elsewhere in general.